### ISSUE TITLE
remove_foreign_key should raise and give instructions

### ISSUE THREAD

**Issue by PhilCoggins** [ankane/strong_migrations#119]
Hello,

We just experienced some production downtime in our application from removing a foreign key with the `remove_foreign_key` rails migration command. [We believe](https://www.postgresql.org/docs/12/explicit-locking.html) this took an `ACCESS EXCLUSIVE` lock on one of our most frequently-read tables. While we're still investigating, I wanted to at least get the conversation started on getting `remove_foreign_key` added as a critical check for this library. Nobody on our team expected that removing this constraint would cause downtime, but it certaintly happened for us this morning.

For reference, we're on Postgres 12.2 and Rails 5.2.3.

**Comment by ankane** [comment]
Hey @PhilCoggins, let me know what you find (and what instructions you would give).

Most `ALTER TABLE` operations require an `ACCESS EXCLUSIVE` lock, as mentioned [here](https://www.postgresql.org/docs/current/sql-altertable.html), so I don't think this is exclusive to removing foreign keys. Queries can queue behind a statement waiting to acquire a lock, which is why it's important to have a short lock timeout to cancel statements that wait for a lock for too long.

**Comment by PhilCoggins** [comment]
@ankane We ended up adding a custom_check on the `remove_foreign_key` method that warns that an `ACCESS EXCLUSIVE` lock is acquired on both the table with the FK and the referenced table. Our organization has a "large migration" window to run these sorts of migrations only during off-peak hours. We raise on the migration, inform the developer of the risks, tell them to make a plan to run the migration during this large migration window, and then direct them to wrap the statement in `safety_assured` to proceed.

Aside from running this migration during off-peak hours, I can't think of any "safe" ways that this migration could be run. We've added several custom checks now that don't necessarily inform the developer of a better way to run the migration, but rather warn them of the risks of running them and to have them run during off-peak hours as to have minimal impact on our customers.

I'm not sure what criteria you use when determining whether to add a check to your library, and if it requires a better alternative than running the migration during off-peak hours, but I guess in my eyes Rails makes it too easy with `remove_foreign_key` that the majority of developers who use it will not understand the risks and may experience downtime just as we did yesterday. A simple warning could help prevent future organizations from going through the same pain that we did. 

The lock timeout is a good suggestion, we have not configured that and will be exploring that. Thank you for the suggestion and the great library!

**Comment by ankane** [comment]
I just pushed out a new version to address this, since I think the importance of a lock timeout was understated in the docs, with the following changes:

1. The installation instructions now generate an initializer with a short lock timeout by default
2. Strong Migrations checks for a short lock timeout in non-dev/test environments and warns if one is not detected
3. The readme has been updated with stronger wording

This should make things safer for everyone.
